### DISCUSSION TITLE
强制开启智能防挡弹幕

### DISCUSSION THREAD

**Discussion by ZiSaAcZ7** [the1812/Bilibili-Evolved#2624] [功能建议 / Ideas]
不知道啊b的智能防挡弹幕是怎么实现的，是有云端预计算再把蒙版发到本地么？只要视频一卡就会显示检测到视频卡顿然后关闭智能防挡弹幕。想要一个强制保持开启的功能捏。

**Comment by the1812** [comment]
应该是本地计算的, 加上蒙版需要耗费很多性能, 所以视频卡顿会自动关

**Reply by ZiSaAcZ7** [reply]
问题是我是5950x+3090，不管他是啥计算我都不至于不够，我的卡顿大部分是网络卡。。

**Reply by the1812** [reply]
那看来 b 站把网络卡和性能卡混一起了, 有测试视频吗

**Reply by ZiSaAcZ7** [reply]
啊？测试视频？只要有这个功能的视频我都会出这个问题

**Reply by the1812** [reply]
好的

**Reply by ZiSaAcZ7** [reply]
卧槽，我今天发现是错怪啊b了，这玩意本地计算蒙版是真的需求巨大，今天有一个视频没有自动关闭防挡弹幕直接卡的飞起了，一看CPU占用好家伙50%。看来还真是本地计算机能不够给我关了。。
测试视频https://www.bilibili.com/video/BV1tR4y1M7rN

**Reply by the1812** [reply]
主要占 GPU 吧, 你是不是没开硬件加速啥的

开启蒙版
![image](https://user-images.githubusercontent.com/26504152/149857124-5a1f05e0-3ea1-4eff-aad2-f5098084a76d.png)
关闭蒙版
![image](https://user-images.githubusercontent.com/26504152/149856903-e4c410eb-e5a9-49b4-aeb3-8422da67eec5.png)

不过即便是这样, 我来回切换下标签页阿 b 直接认为丢帧给关了

**Comment by Kosette** [comment]
检测视频卡顿这个功能，多点点快进也会提示
